source/linux/Foundational_Components/System_Security/SELinux.rst

```diff
@@ -0,0 +1,141 @@
+
+- Enforcing mode is the recommended mode of operation where SELinux operates normally, enforcing the loaded security policy on the entire system.
+- In permissive mode, the system acts as if SELinux is enforcing the loaded security policy, including labeling objects and emitting access denial entries in the logs, but it does not actually deny any operations. While not recommended for production systems, permissive mode can be helpful for SELinux policy development and debugging.
+- Disabled mode is strongly discouraged; not only does the system avoid enforcing the SELinux policy, it also avoids labeling any persistent objects such as files, making it difficult to enable SELinux in the future.
+
+.. note::
+
+   This guide assumes that the user understands how to build the Linux Kernel and, Yocto or Debian distribution.
+
+Filesystem Setup
+----------------
+
+Yocto
+^^^^^
+
+The following steps describe how to build SELinux user-space tools and configuration on Yocto. Please use :ref:`Processor SDK - Building the SDK with Yocto <building-the-sdk-with-yocto>` as reference.
+
+1. On a fresh build, update the config file to use :file:`configs/arago-scarthgap-selinux-config.txt`.
+
+   .. note::
+
+      The default policy is targeted. If you want to update it, add the variable PREFERRED_PROVIDER_virtual/refpolicy = "refpolicy-<your_choice>" where <your_choice> is targeted/mls/mcs in your local_conf file
+
+2. Build your custom image in build directory
+
+   .. code-block:: console
+
+      $ MACHINE=<machine> bitbake -k tisdk-default-image
+
+3. Flash the generated wic image onto your SD card. By default, SELinux functionality is in the disabled state
+
+Debian
+^^^^^^
+
+The following steps describe how to install and configure SELinux together with the default policy which enables SELinux for the most important parts of your Debian filesystem. This guide does not describe how to build the filesystem using Debian.
+
+1. The default Debian image provided by TI packages necessary kernel configs to enable SELinux. If you are not using the standard linux kernel provided by debian with an ext2/3/4/xfs/btrfs filesystem, you have to make sure that you are using a SELinux capable kernel and filesystem. For ext2/3/4 there is nothing special to be aware of, but for other filesystems there are still some quirks.
+
+2. On the target machine, get the default policy and the basic set of SELinux utilities by running
+
+   .. code-block:: console
+
+      $ sudo apt-get install selinux-basics selinux-policy-default auditd
+
+Permissive mode Setup
+---------------------
+
+It is intended to go into Permissive mode, log the accesses, create the access policy and only then jump to Enforcing mode. Skipping this step would lead to a login failure due to lack of access.
+
+1. Boot the target (in disabled state), Run ``fixfiles -F onboot`` or ``touch /.autorelabel`` to configure GRUB and PAM and to create /.autorelabel.
```

Comment:
```suggestion
1. Boot the target (in disabled state), Run ``fixfiles -F onboot`` or ``touch /.autorelabel``
```

This command does not affect GRUB or PAM. Anything can create that file to trigger relabeling. If you mean that the user may use either of those tools to set that, that's also misleading. GRUB should not be modifying that file. There are different kernel parameters that should be used to control selinux instead.

Reply:
Thanks